exiftool/helper.py

```diff
@@ -188,6 +189,11 @@ def get_tags(self, files, tags, params=None):
 
 			# TODO: this list copy could be expensive if the input is a very huge list.  Perhaps in the future have a flag that takes the lists in verbatim without any processing?
 
+		if check_exist:
+			for file in final_files:
+				if not os.path.exists(file):
```

Comment:
`Path(file).exists()`

Reply:
I don't remember if I've said this before, but I would actually type the function accordingly:

```python
from pathlib import Path

...

def get_tags(self, files: list[Path], tags: list[str], params: list[str]):
    ...
```

Reply:
well... I think ```files: list[Path]``` might be a bit of a stretch... ```Union[list[Path], list[str]]``` I would think, but having Path enforced would questionable.  Say, if check_exist=False, there's no real reason to have it as Path, right?

Reply:
> well... I think `files: list[Path]` might be a bit of a stretch... `Union[list[Path], list[str]]` I would think, but having Path enforced would questionable.

Yes, questionable in a way. However, consider this: You are a user of `pyexiftool`, you import `exiftool` and initialize an instance of the helper, and you get going handing files over to it for processing so you can do something with the tags you get. If you are expected to provide `Path`s instead of strings that represent paths, and something goes wrong, errors originate outside of and before any calls to `exiftool`. This should make it super easy to understand what's going on.

On the other hand, it is common (is it not?) in Python to accept files (as in file handles originating from e.g. `open(...)`) as well as paths as `str`s as well as paths as `Path`s... Since we do not handle streams of data or, generally, anything in memory, but instead pass on paths to `exiftool` (the Perl application), and because I believe in explicit information over implicit guesses, this makes me prefer a `Path`-only approach even more. The interface is not a common one, so it should not appear like one.

> Say, if check_exist=False, there's no real reason to have it as Path, right?

I consider the issues distinct. `Path` ensures that valid paths (dependant on the operating system) come in, *not* that these paths point to existing files.

Reply:
It just hit me... we can't use ```Path``` for this argument.  ```*.jpg``` is a valid argument to ```files```.  But it neither passes the check of ```Path("*.jpg").exists()``` nor is it really a valid ```Path```.

Therefore the ```check_exists``` functionality also doesn't really fit the valid use cases very well... as we would have to do like ```Path.glob(files)``` and I'm sure the glob rules for ```files``` in ```ExifToolHelper``` would differ than that of the perl exiftool

Reply:
> It just hit me... we can't use `Path` for this argument. `*.jpg` is a valid argument to `files`. But it neither passes the check of `Path("*.jpg").exists()` nor is it really a valid `Path`.
> 
> Therefore the `check_exists` functionality also doesn't really fit the valid use cases very well... as we would have to do like `Path.glob(files)` and I'm sure the glob rules for `files` in `ExifToolHelper` would differ than that of the perl exiftool

This is a good point, **however**... 😄

While it is great that `exiftool` (the perl application) supports its own kind of globbing, I do not think that this kind of functionality should be available through the `ExifToolHelper`. In general, if someone wants to use `exiftool` and they have it installed, they can run it from the command line. There is no need for Python or `pyexiftool` in that case.

Of course, one might need to do something similar from within Python. Anything is still possible through `ExifTool`; `ExifToolHelper`, on the other hand, provides convenient functionality that commonly makes sense for Python scripts/applications. This is why existence checks are useful.

Reply:
I "semi-disagree", but let me think about it a bit more... (I'm a bit busy this week but later in the week I should have time.)

I definitely think I made a good choice layering ```ExifToolHelper``` on top of ```ExifTool``` for reasons such as this, but ... hmm.

Probably I'll add some of the ```Exception``` stuff first, then re-visit this after the v0.5.0 gets pushed out (... and then users submit a bunch of issues because it's nothing like the v0.4.x... lol)